sale_order_secondary_unit/tests/test_sale_order_secondary_unit.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.tests import SavepointCase
+
+
+class TestSaleOrderSecondaryUnit(SavepointCase):
+    at_install = False
+    post_install = True
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_uom_kg = cls.env.ref('product.product_uom_kgm')
+        cls.product_uom_unit = cls.env.ref('product.product_uom_unit')
+        cls.product = cls.env['product.product'].create({
+            'name': 'test',
+            'uom_id': cls.product_uom_kg.id,
+            'uom_po_id': cls.product_uom_kg.id,
+            'secondary_uom_ids': [
+                (0, 0, {
+                    'name': 'unit-700',
+                    'uom_id': cls.product_uom_unit.id,
+                    'factor': 0.7,
+                })],
+        })
+        cls.secondary_unit = cls.env['product.secondary.unit'].search([
+            ('product_tmpl_id', '=', cls.product.product_tmpl_id.id),
+        ])
+        cls.product.sale_secondary_uom_id = cls.secondary_unit.id
+        cls.partner = cls.env['res.partner'].create({
+            'name': 'test - partner',
+        })
+        so = cls.env['sale.order'].new({
+            'partner_id': cls.partner.id,
+            'order_line': [(0, 0, {
+                'name': cls.product.name,
+                'product_id': cls.product.id,
+                'product_uom_qty': 1,
+                'product_uom': cls.product.uom_id.id,
+                'price_unit': 1000.00,
+            })],
+            'pricelist_id': cls.env.ref('product.list0').id,
+        })
+        so.onchange_partner_id()
+        cls.order = cls.env['sale.order'].create(
```

Comment:
@sergio-teruel here use a central variable because it uses much time.

Reply:
This is not time consuming, but a convenience for not writing long expression. Not blocking any way.